subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -71,9 +89,192 @@ static int cmd_send(const struct shell *sh, size_t argc, char **argv)
+{
+	
+	struct lwm2m_ctx *ctx = lwm2m_rd_client_ctx();
+	if (!ctx) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	if (argc < 2) {
+		shell_error(sh, "no arguments or path(s)\n");
+		shell_help(sh);
+		return -EINVAL;
+	}
+
+	const char* dtype = "-s"; /* default */
+	if (argc > 2) { /* read + path + options(data type) */
+		dtype = argv[2];
+	}
+
+	const char* pathstr = argv[1];
+	int ret = 0;
+	
+	if (strcmp(dtype, "-s") == 0){
+		const char* buff;
+		uint16_t buff_len=0;
+		uint8_t buff_flags;
+		ret = lwm2m_engine_get_res_data(pathstr, (void **)&buff, &buff_len, &buff_flags);
+		shell_print(sh, "%s\n", buff);
+	} else if (strcmp(dtype, "-s8") == 0){
+		int8_t temp=0;
+		ret = lwm2m_engine_get_s8(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	} else if (strcmp(dtype, "-s16") == 0){
+		int16_t temp=0;
+		ret = lwm2m_engine_get_s16(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	} else if (strcmp(dtype, "-s32") == 0){
+		int32_t temp=0;
+		ret = lwm2m_engine_get_s32(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	} else if (strcmp(dtype, "-s64") == 0){
+		int64_t temp=0;
+		ret = lwm2m_engine_get_s64(pathstr, &temp);
+		shell_print(sh, "%lld\n", temp);
+	} else if (strcmp(dtype, "-u8") == 0){
+		uint8_t temp=0;
+		ret = lwm2m_engine_get_u8(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	} else if (strcmp(dtype, "-u16") == 0){
+		uint16_t temp=0;
+		ret = lwm2m_engine_get_u16(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	} else if (strcmp(dtype, "-u32") == 0){
+		uint32_t temp=0;
+		ret = lwm2m_engine_get_u32(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	} else if (strcmp(dtype, "-u64") == 0){
+		uint64_t temp=0;
+		ret = lwm2m_engine_get_u64(pathstr, &temp);
+		shell_print(sh, "%lld\n", temp);
+	} else if (strcmp(dtype, "-f") == 0){
+		double temp=0;
+		ret = lwm2m_engine_get_float(pathstr, &temp);
+		shell_print(sh, "%f\n", temp);
+	} else if (strcmp(dtype, "-b") == 0){
+		bool temp; 
+		ret = lwm2m_engine_get_bool(pathstr, &temp);
+		shell_print(sh, "%d\n", temp);
+	}else {
+		shell_error(sh, "can't recognize data type %s\n", dtype);
+		shell_help(sh);
+		return -EINVAL;
+	}
+	
+	if (ret < 0) {
+		shell_error(sh, "can't do read operation, request failed\n");
+		return ret;
+	}
+	return 0;
+}
+
+static int cmd_write(const struct shell *sh, size_t argc, char **argv) 
+{
+	struct lwm2m_ctx *ctx = lwm2m_rd_client_ctx();
+	if (!ctx) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+	if (argc < 3) {
+		shell_error(sh, "no arguments or path(s)\n");
+		shell_help(sh);
+		return -EINVAL;
+	}
+
+	int ret = 0;
+	const char* pathstr = argv[1];
+
+	const char* dtype = "-s"; /* default */
+	if (argc == 4) /* write path options value */
+	{
+		dtype = argv[2]; 
+	}
+
+	const char* value = argv[3];
+	char* e;
+	if (strcmp(dtype, "-s") == 0){
+		ret = lwm2m_engine_set_string(pathstr, argv[2]);
+	} else if (strcmp(dtype, "-s8") == 0){
```

Comment:
```suggestion
	} else if (strcmp(dtype, "-s8") == 0) {
```
Fix this on all rest of these as well. Space is missing before the opening bracket.